### ISSUE TITLE
Daemon shouldn't have to be run with `sudo -E` in Linux

### ISSUE THREAD

**Issue by dnwiebe** [MASQ-Project/MASQ-Node-issues#297]
In Linux, if you run a command with `sudo`, the `HOME` environment variable changes to `/root`, which screws up the Daemon's `--data-directory` setting. If you run the command with `sudo -E`, the `$HOME` environment variable stays at `/home/`<username>, which makes things fine for the Daemon.

Fix it so that the Daemon is fine without needing the `-E` option to `sudo`.

**Comment by dnwiebe** [comment]
We found code in `bootstrapper.rs` that looks like it should extract the pre-`sudo` username from `SUDO_USER` and use it to construct a home-directory path on Linux and macOS that should be properly used in the construction of the data directory. Therefore, this card may be obsolete, although it should still be tested to make sure nothing was forgotten.

However, something that _isn't_ mentioned on this card is the difficulty presented by the fact that important configuration can be in `MASQ_XXX` environment variables, and if the Daemon is not started with `sudo -E`, that configuration will be lost.

We investigated and discovered that on Linux, a process can get its parent's environment from the `/proc/`<parent process ID>`/environ` virtual file, after a bit of parsing, but we couldn't find any parallel method on macOS (`procfs`, the system utility that produces the `/proc` virtual directory, doesn't exist on macOS and may not exist on some flavors of Linux).

Therefore, our conclusion is that somebody who is technically savvy enough to set up environment-variable configuration should also be savvy enough to use `sudo -E` instead of `sudo` to start the Daemon. Hence, no further card about this functionality is necessary.
